tools/reviewer/web/src/app/dashboard/reviews-panel/diff/code-block/code-block.component.ts

```diff
@@ -37,7 +37,17 @@ export class CodeBlockComponent implements OnInit, OnDestroy {
   @Input() fileContent: string;
   @Input() isNewCode: boolean;
   @Input() changes: number[];
-  // TODO: Do not use Threads
+  // TODO: add supporting of deleting whole thread.
```

Comment:
Why do we have all these issues? Is it because we're using protobins somehow?

Reply:
Actually no.
The issues were present in the last update already. I thought to fix them by adding a feature to update a specific comment instead of a whole diff.
But now, when I realized that we going to keep proto binary in firebase, I added more info about the issues.

Reply:
Why doesn't changing the diff locally (i.e adding/removing comments and threads) and updating the whole diff to Firestore work?

Reply:
It works! Actually, it's exactly how it implemented here.
We just need to add one new feature. At the moment the app just doesn't know what it should to do, when it gets a message from firebase, that whole thread was deleted.

Two other things are not issues. It's a notice for developers who will be implement deleting of comments.

Reply:
Got it, thanks :)

Reply:
@MateMahinKhan deleting a thread doesn't break proto structure. We aren't deleting all threads, we're just deleting an element.

Reply:
Okay. so using proto binary will change this anyway.